cli/module_registry.go

```diff
@@ -0,0 +1,260 @@
+
+	client, err := NewAppClient(c)
+	if err != nil {
+		return err
+	}
+	manifestPath := defaultManifestFilename
+	if manifestPathArg != "" {
+		manifestPath = manifestPathArg
+	}
+	if _, err := os.Stat(manifestPath); err != nil {
+		return errors.Wrapf(err, "Cannot find %s", manifestPath)
+	}
+
+	manifestBytes, err := os.ReadFile(manifestPath)
+	if err != nil {
+		return err
+	}
+	var manifest ModuleManifest
+	if err := json.Unmarshal(manifestBytes, &manifest); err != nil {
+		return err
+	}
+	moduleID, err := parseModuleID(manifest.Name)
+	if err != nil {
+		return err
+	}
+	if publicNamespaceArg != "" {
+		switch moduleID.Namespace {
+		case "":
+			moduleID.Namespace = publicNamespaceArg
+		case publicNamespaceArg:
+			// the meta.json manifest == public_namespace arg
+			fmt.Fprintf(c.App.Writer, "The module's %s already specifies a public namespace. Ignoring\n", defaultManifestFilename)
+		default:
+			// the meta.json manifest != public_namespace arg
+			// we may want to investigate a better way of handling this error case
+			// For now, it seems like bad UX to ignore this error
+			return errors.Errorf("the module's %s specifies a namespace of '%s'"+
+				" but a namespace of '%s' was provided in command line arguments",
+				defaultManifestFilename, moduleID.Namespace, publicNamespaceArg)
+		}
+	}
+	var orgID *string
+	if orgIDArg != "" {
+		if moduleID.Namespace == "" {
+			orgID = &orgIDArg
+		} else {
+			fmt.Fprintf(c.App.Writer, "A public namespace (%s) is specified in the config."+
+				" It is not necessary to provide an org_id\n", moduleID.Namespace)
+		}
+	}
+	if orgID == nil && moduleID.Namespace == "" {
+		return errors.Errorf("The module's namespace is not set in the %s."+
+			" You must provide a public_namespace (if you have set one) or supply your org id", defaultManifestFilename)
+	}
+	manifest.Name = moduleID.toString()
+
+	response, err := client.UpdateModule(manifest, orgID)
+	fmt.Fprintf(c.App.Writer, "Module successfully updated! You can view your changes online here: %s\n", response.GetUrl())
+	if err != nil {
+		return err
+	}
+
+	// If the namespace isn't set, modify the meta.json to set it (if available)
+	if moduleID.Namespace == "" {
+		org, err := resolveOrg(client, orgIDArg, publicNamespaceArg)
+		if err != nil {
+			return err
+		}
+		if org.PublicNamespace != "" {
+			moduleID.Namespace = org.PublicNamespace
+			manifest.Name = moduleID.toString()
+			if err := writeManifest(manifestPath, manifest); err != nil {
+				return err
+			}
+			fmt.Fprintf(c.App.Writer, "\nUpdating %s to use the public namespace of %q which is %q\n",
+				manifestPath, org.Name, org.PublicNamespace)
+			fmt.Fprintf(c.App.Writer, "You no longer need to specify org_id or public_namespace\n")
+		}
+	}
+
+	return nil
+}
+
+func parseModuleID(moduleName string) (ModuleID, error) {
+	// This parsing is intentionally lenient so that the backend does the real validation
+	splitModuleName := strings.Split(moduleName, ":")
```

Comment:
I actually like keeping this instead of regex and during testing we can see if there are any issues.